### DISCUSSION TITLE
User Feedback / Progress when importing Beatmaps from previous install

### DISCUSSION THREAD

**Discussion by OfficialCRUGG** [ppy/osu#25485] [Ideas]
On the first-run setup, I had imported my ~600 beatmaps from an old build of osu!lazer. (Attached an image of the screen in question below). This whole process took about 3-4 minutes and since it only had an indeterminate loading spinner, I wasn't sure if it glitched out or something. I would propose having some sort of progress indication to avoid user confusion and improve UX.

![image](https://github.com/ppy/osu/assets/25248999/a70617dc-ab72-4e69-a235-3d1c1544deb7)

**Comment by peppy** [comment]
That screen lets you start an import but the idea of that you don't have to wait at it. If you continue, you'll end up with a proper progress bar as a notification.

![CleanShot 2023-11-17 at 04 47 25](https://github.com/ppy/osu/assets/191335/cf49e06b-062a-47b6-8e47-283adbf94fc6)


I agree that the flow of the first run dialog needs some improvement though

**Reply by OfficialCRUGG** [reply]
If I'm being honest, I completely overlooked it. I've done some research now and understand why it takes this long, but at first my instinct just was that importing some files I already have on my hard drive should take no more than a few seconds. So this was very unexpected to me.
